plone/namedfile/tests/test_storable.py

```diff
@@ -56,3 +58,47 @@ def test_opened_file_storable(self):
             if os.path.exists(path):
                 os.remove(path)
         self.assertEqual(303, fi.getSize())
+
+    def test_upload_no_read(self):
+        # ensure we don't read the whole file into memory
+
+        import ZODB.blob
+
+        old_open = ZODB.blob.Blob.open
+        blob_read = 0
+        blob_write = 0
+
+        def count_open(self, mode="r"):
+            nonlocal blob_read, blob_write
+            blob_read += 1 if "r" in mode else 0
+            blob_write += 1 if "w" in mode else 0
+            return old_open(self, mode)
+
+        ZODB.blob.Blob.open = count_open
+
+        data = getFile("image.gif")
+        f = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            path = f.name
+            f.write(data)
+            f.close()
+            with open(path, "rb") as f:
+                fi = NamedBlobFile(f, filename="image.gif")
+        finally:
+            if os.path.exists(path):
+                os.remove(path)
+        self.assertEqual(303, fi.getSize())
+        self.assertEqual(blob_read, 1, "blob should have only been opened to get size")
+        self.assertEqual(
+            blob_write,
+            1,
+            "Slow write to blob instead of os rename. Should be only 1 on init",
+        )
+        blob_read = 0
+
+        blob_field = field.NamedBlobFile()
+        blob_field.validate(fi)
+
+        self.assertEqual(blob_read, 0, "Validation is reading the whole blob in memory")
+
+        ZODB.blob.Blob.open = old_open
```

Comment:
Clever test. This last line should go in a `finally` block so that you don't leak the patched method if there's an error in the test. (Or it might also be possible to use unittest.mock.patch.object as a context manager)